### DISCUSSION TITLE
Radio Navigation (old style)

### DISCUSSION THREAD

**Discussion by bagolu** [albar965/littlenavmap#1030] [General]
I would like to use LNM with radio navigation : The 1944 station list is available as a CSV file.
It includes the location, a radio frequency (kHz) and 4 beam directions.

Navigating with theses beams is done listening to audio transmission of morse code (A & N)
A simulator addon exists : https://flightsim.to/file/21201/radio-range-navigation

What's missing with LNM is a representation of the beam direction like ILS (Beam and N side)
https://cdn.flightsim.to/images/18/radio-range-navigation-_eMZv.jpg

Does anyone thinks this could be done in a way or another ?

**Comment by albar965** [comment]
It could be done by extending the CSV file format with a bearing number. 
Do you have an example or link for such CSV file?

**Comment by bagolu** [comment]
In the radio range navigation mod there are 2 csv : 

One specific for LNM. Basic format (without beam directions) :
TACAN            AP 379 32.475  -99.6833        0
TACAN            AK 371 35.05   -107.6689       0
TACAN            AF 317 37.0775 -89.9217        0

And one with the directions : 
2       371     Acomita LFR     AK      35.05   -107.6689       A       87      177     267     357
3       317     Advance LFR     AF      37.0775 -89.9217        A       81      181     261     342
4       368     Akron Colorado  LFR     RN      40.1625 -103.1878       A       78      168     258     348

Columns from this one are : 
id,freq,name,morse code,lat,long,A or N type,dir1,dir2,dir3,dir4

But how would they be represented in LNM ? 
I haven't found any userpoint with such capability

**Reply by albar965** [reply]
> But how would they be represented in LNM ? I haven't found any userpoint with such capability

Currently there is none. I'd have to add this as a new feature for a future release. I'd have to add the bearing option as a new column in the CSV. This can be added while maintaining backwards compatibility.

**Reply by bagolu** [reply]
Can this be integrated like other nav aids (vor, ndb...) ?

**Reply by bagolu** [reply]
![image](https://github.com/albar965/littlenavmap/assets/39214698/2ca862d6-2e7a-44cb-8218-f819d881088e)
It seems someone managed to do something. Icons seem to be userpoints but I don't know how he made the beams

**Reply by albar965** [reply]
> Can this be integrated like other nav aids (vor, ndb...) ?

Sorry, too much effort. Easiest way is to add these as userpoints which can be added to flight plans too.

**Reply by albar965** [reply]
> ![image](https://user-images.githubusercontent.com/39214698/243455895-2ca862d6-2e7a-44cb-8218-f819d881088e.png) It seems someone managed to do something. Icons seem to be userpoints but I don't know how he made the beams

No idea. You have a source or link?

**Reply by bagolu** [reply]
unfortunately no... only this screenshot

userpoints are super fine especially for this kind of oldies flight planning

**Reply by albar965** [reply]
> unfortunately no... only this screenshot
> 
> userpoints are super fine especially for this kind of oldies flight planning

Noted in issue #1031.

**Reply by bagolu** [reply]
Ok, I have the fine word on this : beams are made with user airspaces !

**Reply by albar965** [reply]
Ok. Thanks for the confirmation. I can only add the userpoint icon rotated by the bearing.

**Comment by bagolu** [comment]
Not sure it is usefull : the 4 beams are not at 0,90,180,270 and rotated: they depend on directions to next radio stations
I guess you can ignore my request and leave the userpoint import as it is : we can already set the usefull information as text/comment.
The airspaces is not a bad thing to display the directions.

Of course, if one day you rewrite the other navaids system, maybe this could be taken into account :) 
Thanks for the quick replies !

**Reply by albar965** [reply]
There is still the issue #1031 as a reminder. Have to see what can be done.
